src/test/services/openklant2.service.test.ts

```diff
@@ -0,0 +1,112 @@
+import { extractOnderwerp } from "@/services/openklant2/service";
+import { describe, it, expect } from "vitest";
+import type { Vraag } from "@/stores/contactmoment";
+
+describe("extractOnderwerp", () => {
+  const makeVraag = (overrides = {}): Vraag => ({
+      vraag: { title: "Testvraag", url: "https://example.com" },
+      specifiekevraag: "Specifieke vraag",
+      zaken: [],
+      notitie: "",
+      contactverzoek: {
+          vragenSets: [],
+          vragenSetIdMap: new Map(),
+      },
+      startdatum: "",
+      kanaal: "",
+      ...overrides,
+      gespreksresultaat: "",
+      klanten: [],
+      medewerkers: [],
+      websites: [],
+      kennisartikelen: [],
+      nieuwsberichten: [],
+      werkinstructies: [],
+      vacs: []
+  });
+
+  it("returns specifiekevraag truncated if title is 'anders'", () => {
+    const vraag = makeVraag({ vraag: { title: "anders" }, specifiekevraag: "a".repeat(210) });
+    const result = extractOnderwerp(vraag);
+    expect(result.length).toBeLessThanOrEqual(200);
+    expect(result.endsWith("..."));
+  });
+
+  it("returns title and specifiekevraag if both fit", () => {
+    const vraag = makeVraag({ vraag: { title: "Vraag" }, specifiekevraag: "Specifiek" });
+    const result = extractOnderwerp(vraag);
+    expect(result).toBe("Vraag (Specifiek)");
+  });
+
+  it("truncates title if title + specifiekevraag is too long", () => {
+    const vraag = makeVraag({ vraag: { title: "a".repeat (190) }, specifiekevraag: "Specifiek" });
+    const result = extractOnderwerp(vraag);
+    expect(result.length).toBeLessThanOrEqual(200);
+    expect(result.endsWith(")"));
+    expect(result.includes("..."));
+  });
+
+  it("returns only title if only title is present and fits", () => {
+    const vraag = makeVraag({ specifiekevraag: undefined });
+    const result = extractOnderwerp(vraag);
+    expect(result).toBe("Testvraag");
+  });
+
+  it("truncates title if only title is present and too long", () => {
+    const vraag = makeVraag({ vraag: { title: "a".repeat(210) }, specifiekevraag: undefined });
+    const result = extractOnderwerp(vraag);
+    expect(result.length).toBeLessThanOrEqual(200);
+    expect(result.endsWith("..."));
+  });
+
+  it("returns specifiekevraag truncated if only specifiekevraag is present", () => {
+    const vraag = makeVraag({ vraag: { title: undefined }, specifiekevraag: "a".repeat(210) });
+    const result = extractOnderwerp(vraag);
+    expect(result.length).toBeLessThanOrEqual(200);
+    expect(result.endsWith("..."));
```

Comment:
Be aware, this does nothing!!
It should be something like expect(result.endsWith("...")).toBeTruthy()